### PR TITLE
Add BIP keystore and truststore, add BIP evidence api test endpoint.

### PR DESCRIPTION
**Pull request by wlin2019**
Add BIP CE test endpoint

## What was the problem?
This PR tries to address two problems.
1. BIP TLS credentials;
2. Add an endpoint to test BIP CE file upload api.

Note: some data has been updated in secret repo. Checkout the secret repo and run scripts/setenv.sh before build.

## How to test this PR
- Step 1. check out secret repo, and build the app followed the normal build procedure.
- Step 2. bring up a app, and test get claim and file upload endpoints in swagger UI.
- Step 3. Set ENV=dev, and ./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp
- Step 4. Repeat step 2. the endpoints should respond with error for host not found as your local environment does not have access to BIP API.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by au2 — CHANGES_REQUESTED** [review]
Lint and test are failing.

File: app/src/test/resources/application-test.yml [hunk 1]
```diff
@@ -49,5 +49,7 @@ bip:
   applicationName: "${BIP_APPLICATION_ISS}"
   applicationId: "${BIP_APPLICATION_ID}"
   stationId: "${BIP_STATION_ID}"
-truststore: "bipcert.jks"
-truststore_password: "vropassword"
\ No newline at end of file
+truststore: "${BIP_TRUSTSTORE:}"
```
Comment (dimitri-amida) [inline app/src/test/resources/application-test.yml:52]: Should these be under bip?

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java [hunk 2]
```diff
@@ -179,7 +179,7 @@ public FetchPdfResponse uploadPdf(FetchPdfResponse pdfResponse) {
       bipApiService.uploadEvidence(
           FileIdType.FILENUMBER,
           pdfResponse.getClaimSubmissionId(),
-          new BipFileUploadPayload(),
+          BipFileUploadPayload.builder().build(),
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java:182]: Can you please pass the correct parameters for calling file upload here?

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java [hunk 3]
```diff
@@ -265,8 +250,12 @@ public BipFileUploadResp uploadEvidenceFile(
       HttpEntity<MultiValueMap<String, Object>> httpEntity = new HttpEntity<>(body, headers);
 
       ResponseEntity<String> bipResponse =
-          restTemplate.exchange(url, HttpMethod.POST, httpEntity, String.class);
+          ceRestTemplate.exchange(url, HttpMethod.POST, httpEntity, String.class);
       BipFileUploadResp resp = new BipFileUploadResp();
+      log.info(
+          "bip response for upload: status: {}, message: {}",
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java:256]: We need a claim or collection Id in the message to know which claim it is referring to.

File: api/src/main/java/gov/va/vro/api/responses/BipFileUploadResponse.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import io.swagger.v3.oas.annotations.media.Schema;
 import lombok.Builder;
 import lombok.Getter;
+import lombok.Setter;
```
Comment (dimitri-amida) [inline api/src/main/java/gov/va/vro/api/responses/BipFileUploadResponse.java:6]: This setter is not used

File: controller/src/main/java/gov/va/vro/controller/BipController.java [hunk 5]
```diff
@@ -167,4 +157,60 @@ public ResponseEntity<BipContentionCreationResponse> createContentions(
...
+      FileIdType type = FileIdType.getIdType(fileidtype);
+      if (type == null) {
+        BipFileUploadResponse badResp =
+            BipFileUploadResponse.builder().message("Invalid ID type: " + fileidtype).build();
+        return ResponseEntity.badRequest().body(badResp);
+      }
+      BipFileProviderData providerData =
+          BipFileProviderData.builder()
+              .contentSource("VBMS")
+              .claimantFirstName("John")
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/BipController.java:175]: Should all these be hard-coded?

**Review by dimitri-amida — CHANGES_REQUESTED** [review]
@wlin2019 can you please populate BipFileUploadPayload in BipClaimService with the correct parameters for an upload to succeed?

File: api/src/main/java/gov/va/vro/api/resources/BipResource.java [hunk 6]
```diff
@@ -1,10 +1,6 @@
 package gov.va.vro.api.resources;
 
-import gov.va.vro.api.responses.BipClaimContentionsResponse;
-import gov.va.vro.api.responses.BipClaimResponse;
-import gov.va.vro.api.responses.BipClaimStatusResponse;
-import gov.va.vro.api.responses.BipContentionCreationResponse;
-import gov.va.vro.api.responses.BipContentionUpdateResponse;
+import gov.va.vro.api.responses.*;
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/resources/BipResource.java:3]: Do not use "*" form please. That is one the style check. Ian posted a fix for IntelliJ not to do this automatically if that is what you are using.

File: api/src/main/java/gov/va/vro/api/resources/BipResource.java [hunk 7]
```diff
@@ -180,4 +172,34 @@ ResponseEntity<BipContentionCreationResponse> createContentions(
...
+            content = @Content(schema = @Schema(hidden = true))),
+        @ApiResponse(
+            responseCode = "500",
+            description = "Data Access Server Error",
+            content = @Content(schema = @Schema(hidden = true)))
+      })
+  @Timed(value = "bip-evidence-fileupload")
+  @Tag(name = "BIP Integration")
+  ResponseEntity<BipFileUploadResponse> fileUpload(
+      @Parameter(description = "file ID", required = true, schema = @Schema(type = "string"))
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/resources/BipResource.java:196]: So this is all we need for testing? I thought we should have some veteran information or claim information, no?

File: app/src/main/java/gov/va/vro/config/BipApiConfig.java [hunk 8]
```diff
@@ -51,12 +64,51 @@ public BipApiProps getBipApiProps() {
   @Bean(name = "bipCERestTemplate")
   public RestTemplate getHttpsRestTemplate(RestTemplateBuilder builder) throws BipException {
     try { // TODO: keep log for testing, remove it later.
-      log.info("truststore: {}, password: {}", trustStore, password);
-      URL url = getClass().getClassLoader().getResource(trustStore);
+      log.info(
+          "truststore: {}, password: {}, keystore: {}, alias: {}",
+          trustStore.length(),
+          password,
```
Comment (au2) [inline app/src/main/java/gov/va/vro/config/BipApiConfig.java:70]: can you change password to length as well.

File: app/src/test/resources/application-test.yml [hunk 9]
```diff
@@ -49,5 +49,7 @@ bip:
...
+truststore_password: "${BIP_PASSWORD:}"
+keystore: "${BIP_KEYSTORE:}"
+bipalias: "${BIP_ALIAS:}"
```
Comment (au2) [inline app/src/test/resources/application-test.yml:55]: please finish the files with a empty line.

File: controller/src/main/java/gov/va/vro/controller/BipController.java [hunk 10]
```diff
@@ -6,16 +6,8 @@
...
-import gov.va.vro.model.bip.BipUpdateClaimContentionPayload;
-import gov.va.vro.model.bip.BipUpdateClaimPayload;
-import gov.va.vro.model.bip.BipUpdateClaimResp;
-import gov.va.vro.model.bip.ClaimContention;
-import gov.va.vro.model.bip.CreateContention;
-import gov.va.vro.model.bip.CreateContentionReq;
-import gov.va.vro.model.bip.UpdateContention;
-import gov.va.vro.model.bip.UpdateContentionReq;
+import gov.va.vro.api.responses.BipFileUploadResponse;
+import gov.va.vro.model.bip.*;
```
Comment (au2) [inline controller/src/main/java/gov/va/vro/controller/BipController.java:10]: no "*"

File: controller/src/main/java/gov/va/vro/controller/BipController.java [hunk 11]
```diff
@@ -167,4 +157,60 @@ public ResponseEntity<BipContentionCreationResponse> createContentions(
...
+              .benefitTypeId(10)
+              .documentTypeId(131)
+              .dateVaReceivedDocument("2023-01-09")
+              .subject("subject")
+              .contentions(List.of("contention1"))
+              .alternativeDocmentTypeIds(List.of(1))
+              .actionable(false)
+              .associatedClaimIds(List.of("1"))
+              .notes(List.of("[This is a note for a document. These replace editing the summary]"))
+              .payeeCode("00")
```
Comment (au2) [inline controller/src/main/java/gov/va/vro/controller/BipController.java:188]: Why don't we have these fields in the API spec and have these fields as examples? That way we can test with other values.

File: scripts/setenv.sh [hunk 12]
```diff
@@ -140,13 +140,15 @@ exportSecretIfUnset BIP_CLAIM_SECRET
...
 
 # TODO: Move all? of these to application*.yml
 export BIP_CLAIM_URL=claims-uat.stage8.bip.va.gov/api/v1
 export BIP_CLAIM_ISS=virtual_regional_office
 export BIP_EVIDENCE_URL=vefs-claimevidence-uat.stage8.bip.va.gov/api/v1/rest
 export BIP_EVIDENCE_ISS=VRO
 export BIP_APPLICATION_ID=VRO
-export BIP_STATION_ID=281
-export BIP_TRUSTSTORE=bipcert.jks
-export BIP_PASSWORD=vropassword
```
Comment (au2) [inline scripts/setenv.sh:152]: eof please

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java [hunk 13]
```diff
@@ -2,17 +2,7 @@
...
-import gov.va.vro.model.bip.BipContentionResp;
-import gov.va.vro.model.bip.BipFileUploadPayload;
-import gov.va.vro.model.bip.BipFileUploadResp;
-import gov.va.vro.model.bip.BipUpdateClaimResp;
-import gov.va.vro.model.bip.ClaimContention;
-import gov.va.vro.model.bip.ClaimStatus;
-import gov.va.vro.model.bip.CreateContentionReq;
-import gov.va.vro.model.bip.FileIdType;
-import gov.va.vro.model.bip.UpdateContentionReq;
+import gov.va.vro.model.bip.*;
```
Comment (au2) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java:5]: no "*". That is a style check and also clarified what is added with this log.

**Review by au2 — CHANGES_REQUESTED** [review]
See inline

File: app/src/test/resources/application-test.yml (hunk 1 above)
Comment (wlin2019) [reply app/src/test/resources/application-test.yml:52]: Good catch! I'm still testing it, so I put it this way for now. When it is tested, I'll put it under BIP. I should also revisit all the BIP environment variable setting after we get credentials for higher environments. Some of them may be shared and do not need to have for each environment.

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java (hunk 2 above)
Comment (wlin2019) [reply service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java]: Added some with notes.

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java (hunk 3 above)
Comment (wlin2019) [reply service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java:256]: BIP upload API doesn't return a claim ID. A list of claim IDs can be put in the payload (see associatedClaimIds in BipFileProviderData) when calls the API

File: controller/src/main/java/gov/va/vro/controller/BipController.java (hunk 5 above)
Comment (wlin2019) [reply controller/src/main/java/gov/va/vro/controller/BipController.java:175]: Good catch! Of course not. This is just a test endpoint, so I put the only set of data that I can get a positive return in BIP CE UAT swagger ui so far. This endpoint and controller is only for testing the BIP access. In VRO, BipApiService should be used.

File: controller/src/main/java/gov/va/vro/controller/BipController.java (hunk 11 above)
Comment (wlin2019) [reply controller/src/main/java/gov/va/vro/controller/BipController.java:188]: I don't know what are all the valid values. This is the only set of data I got positive response from their UAT swagger UI. So I used it at this test point. In VRO code, the BipApiService should be used. The controller endpoints are only for me to test BIP connection. We don't need it after test is done.

File: api/src/main/java/gov/va/vro/api/resources/BipResource.java (hunk 7 above)
Comment (wlin2019) [reply api/src/main/java/gov/va/vro/api/resources/BipResource.java:196]: Good idea! But I purposely do not include the payload parameter as there are some issues that break the other post endpoints in this controller. I haven't found a fix for that. As the controller is just for testing BIP in dev env, let's get some testing done first and when we have a need to this controller later we can fix it.

File: api/src/main/java/gov/va/vro/api/resources/BipResource.java (hunk 6 above)
Comment (wlin2019) [reply api/src/main/java/gov/va/vro/api/resources/BipResource.java]: Hmm... Intellij did it. Removed *.

File: controller/src/main/java/gov/va/vro/controller/BipController.java (hunk 10 above)
Comment (wlin2019) [reply controller/src/main/java/gov/va/vro/controller/BipController.java]: done

File: app/src/test/resources/application-test.yml (hunk 9 above)
Comment (wlin2019) [reply app/src/test/resources/application-test.yml:55]: Intellij removes it. I'll add it again.

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java (hunk 13 above)
Comment (wlin2019) [reply service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java:5]: done

File: scripts/setenv.sh (hunk 12 above)
Comment (wlin2019) [reply scripts/setenv.sh:152]: Add it.

File: api/src/main/java/gov/va/vro/api/responses/BipFileUploadResponse.java (hunk 4 above)
Comment (wlin2019) [reply api/src/main/java/gov/va/vro/api/responses/BipFileUploadResponse.java:6]: Thanks for pointing out. I forget to clean this up. Will do it later.

**Review by au2 — APPROVED** [review]

**Pull request merged**